ie_serving/models/ir_engine.py

```diff
@@ -40,19 +42,36 @@ def __init__(self, model_xml, model_bin, mapping_config, exec_net,
         logger.info("Matched keys for model: {}".format(self.model_keys))
 
     @classmethod
-    def build(cls, model_xml, model_bin, mapping_config):
+    def build(cls, model_xml, model_bin, mapping_config, batch_size):
         plugin = IEPlugin(device=DEVICE, plugin_dirs=PLUGIN_DIR)
         if CPU_EXTENSION and 'CPU' in DEVICE:
             plugin.add_cpu_extension(CPU_EXTENSION)
         net = IENetwork.from_ir(model=model_xml, weights=model_bin)
-        input_blob = next(iter(net.inputs))
-        batch_size = net.inputs[input_blob].shape[0]
+        if batch_size is not None:

```

Comment:
This whole if-else statement should be moved to separate function. In our unit tests all paths should be tested. Especially with non int input.